clang/test/3C/partial_checked_arr.c

```diff
@@ -2,8 +2,9 @@
 // RUN: 3c -addcr -alltypes %s | %clang -c -f3c-tool -fcheckedc-extension -x c -o %t1.unused -
 // RUN: 3c -addcr %s | FileCheck -match-full-lines -check-prefixes="CHECK_NOALL","CHECK" %s
 // RUN: 3c -addcr %s | %clang -c -f3c-tool -fcheckedc-extension -x c -o %t2.unused -
-// RUN: 3c -alltypes %s > %t
-// RUN: 3c -alltypes %t | count 0
+// RUN: 3c -alltypes -output-postfix=checked %s
+// RUN: 3c -alltypes %S/partial_checked_arr.checked.c -- | count 0
+// RUN: rm %S/partial_checked_arr.checked.c
```

Comment:
This is a regression to when we didn't know about %t temp files. Which admittedly may have been so soon that the changes overlapped. If there's no reason for the regressions, lets leave the simpler code that doesn't create extra files if it ends prematurely.

Reply:
`%t` expanded to a filename ending in `.c.tmp`.  For some reason, `3c` doesn't like that as an input file name and spewed a bunch of errors.  For now, I just changed this test to do the same thing as many of our other tests.  When we have a proper solution to this problem, we can update all the tests.

I'm sorry for not giving you more time to review this.

Reply:
It didn't give me any errors when I committed it. Is that because of tests on a different system, or related to the changes made in this pull request?

Reply:
Before this pull request, when I ran the `3c` command from the `RUN` line manually, I saw the errors, but `3c` still exited 0, so the test passed, and `lit` doesn't seem to show the stderr of passing tests.  With the changes in this pull request, `3c` exited 1, so I had to do something about the test.

Reply:
This seems significant to me, maybe for more issues? One for `.tmp` and one for passing with errors. If 3c ever allows tests to pass that have errors in them, that's bad for everyone. Did this pull request fix all of that? The OP was concerned with a subset and then you fixed more. 

I guess for now we need to remember not to pass lit temp files to 3c. I'll see if there's a good place to fit that in the temp files issue I made earlier to document future changes.

Reply:
> This seems significant to me, maybe for more issues? One for `.tmp` and one for passing with errors.

Agreed!

> If 3c ever allows tests to pass that have errors in them, that's bad for everyone. Did this pull request fix all of that? The OP was concerned with a subset and then you fixed more.

I'm not sure, and I don't really want to spend more time researching it right now.  Someone else is probably more familiar than I am with all the types of errors that can occur in 3c.

> I guess for now we need to remember not to pass lit temp files to 3c. I'll see if there's a good place to fit that in the temp files issue I made earlier to document future changes.

Good.  If we pass a temp file, the test will fail, and we just need to learn quickly that it is because of the temp file and not waste time debugging again.